psi4/extras.py

```diff
@@ -154,12 +154,38 @@ def addons(request=None):
 
 
 # Testing
-def test():
-    """Runs a smoke test suite through pytest."""
+def test(extent='smoke', extras=None):
+    """Runs a smoke test suite through pytest.
 
+    Parameters
+    ----------
+    extent : {'smoke', 'quick', 'full', 'long'}
+        All choices are defined, but choices may be redundant in some projects.
+        _smoke_ will be minimal "is-working?" test(s).
+        #_quick_ will be as much coverage as can be got quickly, approx. 1/3 tests.
+        #_full_ will be the whole test suite, less some exceedingly long outliers.
+        #_long_ will be the whole test suite.
+    extras : list
+        Additional arguments to pass to `pytest`.
+
+    Returns
+    -------
+    int
+        Return code from `pytest.main()`. 0 for pass, 1 for fail.
+
+    """
     try:
         import pytest
     except ImportError:
         raise RuntimeError('Testing module `pytest` is not installed. Run `conda install pytest`')
     abs_test_dir = os.path.sep.join([os.path.abspath(os.path.dirname(__file__)), "tests"])
-    pytest.main(['-rws', '-v', '--capture=sys', abs_test_dir])
+
+    command = ['-rws', '-v']
+    if extent.lower() in ['smoke', 'quick', 'full', 'long']:
```

Comment:
I like this little function a lot!

Reply:
Thanks! I've put it in 3 repos in the past few days and it's destined for a 4th. One just has to remember to `sys.exit(mod.test())` to get it to actually trigger CI failure.